flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java

```diff
@@ -272,6 +272,7 @@ public StreamExecutionEnvironment setMaxParallelism(int maxParallelism) {
 	 * value.
 	 */
 	public int getParallelism() {
+		consolidateParallelismDefinitionsInConfiguration();
```

Comment:
I think we should call this in the constructor, so that it is called at a well known location and not the just the first time `getParallelism()` is called. There is still some legacy weirdness around this setting, because the `ExecutionConfig` sets this to `PARALLELISM_DEFAULT` but this value is then never used. This is for legacy reasons.

Reply:
Thanks @aljoscha , I updated the PR. I think that your suggestion is correct as it respect the priority order of the different places where the user can specify the parallelism. Let's see what Travis has to say.